kloppy/domain/models/common.py

```diff
@@ -294,6 +302,79 @@ def duration(self):
         return self.end_timestamp - self.start_timestamp
 
 
+class Origin(Enum):
+    """
+    Attributes:
+        TOP_LEFT: Origin at the top left of the field
+        BOTTOM_RIGHT: Origin at the bottom left of the field
+        CENTER: Origin at the center of the field
+    """
+
+    TOP_LEFT = "top-left"
+    BOTTOM_RIGHT = "bottom-right"
+    CENTER = "center"
+
+    def __str__(self):
+        return self.value
+
+
+@dataclass
+class CoordinateSystem(ABC):
+    provider: Provider
+    origin: Origin
+    vertical_orientation: VerticalOrientation
+    normalized: bool
```

Comment:
The `provider`, `origin` and `vertical_orientation` are attributes that are tied to a single - valid - value per class. 

Consider using `abstractmethod` and `property` decorators like here: https://github.com/PySport/kloppy/blob/master/kloppy/domain/models/event.py#L343-L346
This way you can enforce that these properties are defined, but prevent them from being passed with a potential wrong value. You can achieve similar behaviour by using `__post__init__` ( https://docs.python.org/3/library/dataclasses.html#post-init-processing ) but I believe in that case that doesn't make sense as there is only 1 single value that is valid.